warehouse/warehouse.go

```diff
@@ -625,6 +625,9 @@ func (wh *HandleT) createJobs(warehouse warehouseutils.WarehouseT) (err error) {
 		// If it is present do nothing else delete it
```

Comment:
Me and @sivashanmukh had a call yesterday, we discussed mainly about the following:
To have a defaultPriority and reset the priority in case you have a waiting job which is !inProgress.
 
 so
```
defaultPriority := 0
uploadID, uploadStatus, priority := wh.getLatestUploadStatus(&warehouse)
if uploadStatus == Waiting {
    if _, inProgess := wh.isUploadJobInProgress(warehouse, uploadID); !inProgess {
		wh.deleteWaitingUploadJob(uploadID)
        defaultPriority = priority
    }
}

...

wh.createUploadJobsFromStagingFiles(warehouse, whManager, stagingFilesList, defaultPriority, uploadStartAfter)

```

Reply:
Wdyt ? @dhawal1248

Reply:
> Me and @sivashanmukh had a call yesterday, we discussed mainly about the following: To have a defaultPriority and reset the priority in case you have a waiting job which is !inProgress.
> 
> so
> 
> ```
> defaultPriority := 0
> uploadID, uploadStatus, priority := wh.getLatestUploadStatus(&warehouse)
> if uploadStatus == Waiting {
>     if _, inProgess := wh.isUploadJobInProgress(warehouse, uploadID); !inProgess {
> 		wh.deleteWaitingUploadJob(uploadID)
>         defaultPriority = priority
>     }
> }
> 
> ...
> 
> wh.createUploadJobsFromStagingFiles(warehouse, whManager, stagingFilesList, defaultPriority, uploadStartAfter)
> ```

+1

Reply:
Makes sense @abhimanyubabbar